saul_cord_ep.c

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ * @ingroup     pkg
+ * @{
+ *
+ * @file
+ * @brief       Register SAUL registry to resource directory
+ *
+ * @author      Matthias Bräuer <matthias.braeuer@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include "saul_cord_ep.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/* stack configuration */
+#define STACKSIZE           (THREAD_STACKSIZE_DEFAULT)
+#define PRIO                (THREAD_PRIORITY_MAIN - 1)
+#define TNAME               "cord_ep"
+
+#define UPDATE_TIMEOUT      (0xe537)
+
+#define TIMEOUT_US          ((uint64_t)(CORD_UPDATE_INTERVAL * US_PER_SEC))
+
+static char _stack[STACKSIZE];
+
+static xtimer_t _timer;
+static kernel_pid_t _runner_pid;
+static msg_t _msg;
+
+static saul_cord_ep_cb_t _cb = NULL;
+
+// The ipv6 address (and port) of the resource directory
+static const char *ip = NULL;
+
+static void _set_timer(void)
+{
+    xtimer_set_msg64(&_timer, TIMEOUT_US, &_msg, _runner_pid);
+}
+
+/**
+ * @brief Execute the callback function if present.
+ *
+ * @param[in] event The event that occurred
+ */
+static void _notify(saul_cord_ep_event_t event) {
+    if(_cb != NULL) {
+        _cb(event);
+    }
+}
+
+/**
+ * @brief Create an UDP socket.
+ *
+ * @param[out] ep Will contain the enpoint of the IPv6 address
+ * @param[in] addr The IPv6 address to be used for the socket
+ * @return `0` on success, else `-1`
+ */
+static int make_sock_ep(sock_udp_ep_t *ep, const char *addr)
+{
+    ep->port = 0;
+    if (sock_udp_str2ep(ep, addr) < 0) {
+        return -1;
+    }
+    ep->family  = AF_INET6;
+    ep->netif   = SOCK_ADDR_ANY_NETIF;
+    if (ep->port == 0) {
+        ep->port = COAP_PORT;
+    }
+    return 0;
+}
+
+/**
+ * @brief Try to register to resource directory service.
+ */
+static int register_rd(void) {
+    sock_udp_ep_t remote;
+    char *regif = NULL;
+
+    if (make_sock_ep(&remote, ip) < 0) {
+        puts("error: unable to parse address\n");
+        return 1;
+    }
+     
+    puts("Registering with RD now, this may take a short while...");
+    
+    if (cord_ep_register(&remote, regif) != CORD_EP_OK) {
+	_notify(SAUL_CORD_EP_REGISTERED);
+        return 1; 
+    }
+    else {
+	_notify(SAUL_CORD_EP_DEREGISTERED);
```

Comment:
Please, indent by 4 spaces (same a few lines above), according to the [RIOT coding conventions](https://github.com/RIOT-OS/RIOT/blob/master/CODING_CONVENTIONS.md).